src/main/java/org/isf/medicalinventory/gui/InventoryBrowser.java

```diff
@@ -412,14 +413,44 @@ private JButton getPrintButton() {
 		jButtonPrint.setEnabled(false);
 		return jButtonPrint;
 	}
-	
+
 	private JButton getDeleteButton() {
 		jButtonDelete = new JButton(MessageBundle.getMessage("angal.common.delete.btn"));
 		jButtonDelete.setMnemonic(MessageBundle.getMnemonic("angal.common.delete.btn.key"));
 		jButtonDelete.setEnabled(false);
+
+		jButtonDelete.addActionListener(actionEvent -> {
+			if (jTableInventory.getSelectedRowCount() > 1) {
+				MessageDialog.error(this, "angasl.inventory.pleaseselectonlyoneinventory.msg");
+				return;
+			}
+			int selectedRow = jTableInventory.getSelectedRow();
+			if (selectedRow == -1) {
+				MessageDialog.error(this, "angal.inventory.pleaseselectinventory.msg");
+				return;
+			}
+			MedicalInventory inventory = inventoryList.get(selectedRow);
+			if (inventory.getStatus().equalsIgnoreCase(InventoryStatus.validated.toString()) ||
```

Comment:
why not extract 1inventory.getStatus()` once and use it instead of multiple `getStatus()` calls?

Reply:
I have done it because it is just two check but I have apply you suggestion